### ISSUE TITLE
"Tank_F" and "Wheeled_APC_F" Invisible Internal Rack(s) Issue

### ISSUE THREAD

**Issue by Quailsnap** [IDI-Systems/acre2#866]
**Mods:**
- **Arma 3:** current stable steam windows as of 24-nov-19
- **CBA:** current steam workshop release as of 24-nov-19
- **ACRE2:** current steam workshop release as of 24-nov-19

**Description:**
Vehicles inheriting from Tank_F and Wheeled_APC_F have inaccessible internal radio racks.

**Steps to Reproduce:**
1. Load CBA_A3, ACE3, ACRE2 in that order.
2. Open a new mission on Stratis. Place down a Rifleman and a Zeus module.
3. Enter the mission.
4. Spawn in any tank that inherits from Tank_F or APC that inherits from Wheeled_APC_F.
5. Tell the AI to get out with "G" or whatever your keybind might be.
6. Get in yourself.
7. There will be no visible radio rack in the vehicle.

**Expected Behavior:**
https://github.com/IDI-Systems/acre2/blob/master/addons/sys_rack/CfgVehicles.hpp#L176
... suggests that both Tank_F and Wheeled_APC_F should start with accessible radio racks.

**Where did the issue occur?**
- Singleplayer.

**Log Files:**
- RPT: https://pastebin.com/8Tn8UBPd

**Additional Context:**
- If you spawn an empty Merkava in the scenario above with Zeus, then run `[vehicle player] call acre_sys_rack_fnc_getVehicleRacks` to **check if any racks are present**, it will return `["acre_vrc103_id_1"]`.
- If you spawn an empty Merkava in the scenario above with Zeus, then run  `[(vehicle player), ["ACRE_VRC103", "Upper Dash", "Dash", false, ["inside"], [], "ACRE_PRC117F", [], ["intercom_1"]], false] call acre_api_fnc_addRackToVehicle` to attempt to **add a new internal rack mid-mission** to the tank, it will add it, but just like the first it will be invisible to ACE self-interact in every single seat.
- If you spawn an empty Merkava in the scenario above with Zeus, then run `[cursorTarget] call acre_api_fnc_areVehicleRacksInitialized` while looking at the tank to ascertain **whether the racks are initialized before any player(s) get in**, it will return `false`, as it should. Upon getting in, running `[(vehicle player)] call acre_api_fnc_areVehicleRacksInitialized` will return `true`, as it should.

**Screenshots:**
N/A

**Comment by Quailsnap** [comment]
Tested again today, still happening as of v2.7.2.1022

**Comment by TheMagnetar** [comment]
It is not that you do not have accessible racks, it is that using the API function you *may* delete the racks. I will recheck.

**Comment by BAVerrecus** [comment]
If I may, I might have found something (as this issue seemd to not have been resolved yet.
The culprit seems to be the line "intercom[] = {"intercom_1"};"
Removing the "intercom_1" let's you access the racks as they should.
Encountered this issue when writing a compat config for a mod. Then tested it with only cba, ace and acre and a config where i made said changes.

Maybe this can halp you track down the underlying issue, if you haven't resolved it already anyway.

**Comment by JudahHarvey** [comment]
I can confirm that this is still happening. I cannot see rack radios in the NATO ifv at all.

**Comment by jonpas** [comment]
They work correctly for me, or at least by design. You have to select the radio as receive or transmit on the intercom, as that radio is only connected to the intercom. After that you can configure/open it using ACE self-interaction or regular ACRE keybinds.

Now if that design is good or not is another perspective. @TheMagnetar we talked about this in-depth and I am still not sure if we want to change some logic so intercom-only radios would also appear in vehicle interaction. Right now you can only edit them after switching intercom receive or transmit to it, but shouldn't you be able to configure them before doing that?

**Comment by jonpas** [comment]
@TheMagnetar I found another issue related to this, if what I described above is indeed by design then this probably shouldn't happen: when you have a radio such as described in this vehicle, only accessible through intercom, you can disconnect from said intercom and the radio becomes accessible directly.

What should it be, how should it work?

**Comment by TheMagnetar** [comment]
it becomes visible if you have "direct access" to it, because you are no longer using it through the intercom but using it directly. By direct access i mean a seat where you can configure it (change frequencies and so on)

**Comment by jonpas** [comment]
That part is correct then.

However there is still no way to configure a radio when you are connected to an intercom and should have physical access to it to turn its knobs.

**Comment by Drofseh** [comment]
I think having intercom accessible racks appear in self interaction instead of vehicle interaction is a bad design choice.

It's still a vehicle rack not a personal radio so it should be under racks as happens with rack radios that are not intercom connected.

**Comment by jonpas** [comment]
**Current:**

Disconnected Intercom:
- Can modify **everything** through target-interaction
![](https://i.imgur.com/7ACrmG9.jpg)

Connected Intercom + Work Knob "INT":
- Can modify **nothing** at all anywhere <- **this is the bigger problem**

Connected Intercom + Work Knob "A" and/or Monitor Knob "A":
- Can modify **everything** through self-interaction
- Can modify **nothing** through target-interaction <- **this is the smaller problem**
![](https://i.imgur.com/uvRdb2K.jpg)

**Proposal:**

Disconnected Intercom:
- Stays the same

Connected Intercom + Work Knob "INT":
- Allow modification through target-interaction <- **solves the main problem** and additionally makes it easier to see what the vehicle contains and is capable of radio-wise
  - Except "Use all"/"Stop using all" and "Use"/"Stop using" interactions on all racks that are in-use by intercom

Connected Intercom + Work Knob "A" and/or Monitor Knob "A":
- Additionally allow modifications through target-interaction

---

> I think having intercom accessible racks appear in self interaction instead of vehicle interaction is a bad design choice.

The only bad design choice IMO is that it's only accessible through one interaction and not the other. As racks are otherwise accessible through target-interaction and once you use them they are accessible through self-interaction as well (in addition to target-interaction).

With my change they would act the same, except they'd be automatically "Used" as you are already connected to the intercom.

In short, I agree with you here.

---

One thing to note here, since we don't have a master station (eg. VIC5) yet we can't dynamically connect/disconnect radios. I don't think that logic would change at that point at all, it would only give extra ability to be connected to the intercom and use the radio separately. As right now, you can just disconnect from the intercom and use the radio directly (kind of simulating disconnecting), but someone else may use it through intercom (which is technically not possible?), and there is only either-or. But that's a separate issue, I just wanted to note it so we don't dive too far into this at this time.

**Comment by Drofseh** [comment]
> accessible through one interaction and not the other

I don't mind if they're accessible in both places, but not having them with the other racks has confused a lot of people in my unit. I had to make an instructional video.

**Comment by TheMagnetar** [comment]
you have to remove the PTT option at least. otherwise it will break the monitor/work knob functionaltiy.

**Comment by jonpas** [comment]
> you have to remove the PTT option at least. otherwise it will break the monitor/work knob functionaltiy.

I fail to see how that would break it. You can only switch PTT to another keybind. The only way to remove it would be to have 4 radios in total and set all the others to some PTT so intercom-connected one is left without one. And even that, knobs wouldn't be broken, you would just not have a key to speak on that radio. :thinking: 

PTT option was also always available from self-interaction menu.

---

I am not sure, but you may mean it would break if in this case: Intercom Work Knob on "INT" and then selecting a PTT on it, it would effectively put it "In use". But that's not possible until the radio is selected using the Work Knob anyways.

You can get only in 1 issue that has already been solved by you: Work Knob on "INT" and Monitor Knob on "A". In this case you get those actions (including PTT assignment) available, and it will actually (currently on last release) by default set to whatever next available PTT button is. If you press it you just get a nice "Radio rack disabled for transmitting in intercom." notification. :)

So I see absolutely no problem here, and I've tried a lot of cases at this point.
